packages/quantic/playwright/page-object/searchObjectWithDidYouMean.ts

```diff
@@ -0,0 +1,91 @@
+import {Page} from '@playwright/test';
+import {SearchObject} from './searchObject';
+
+export type WordCorrectionData = {
+  correctedWord: string;
+  originalWord: string;
+  length: number;
+  offset: number;
+};
+
+export type DidYouMeanData = {
+  correctedQuery: string;
+  wordCorrections: WordCorrectionData[];
+};
+
+export type DidYouMeanNextData = {
+  correctedQuery: string;
+  wordCorrections: WordCorrectionData[];
+  originalQuery: string;
+};
+
+export type QueryTriggerData = {
+  type: string;
+  content: string;
+};
+
+export class SearchObjectWithDidYouMeanOrTrigger extends SearchObject {
+  constructor(page: Page, searchRequestRegex: RegExp) {
+    super(page, searchRequestRegex);
+  }
+
+  async mockSearchWithDidYouMeanResponse(didYouMeanDataObject: DidYouMeanData) {
```

Comment:
To emphasize the difference between the legacy and the next function. 
```suggestion
  async mockSearchWithDidYouMeanLegacyResponse(didYouMeanDataObject: DidYouMeanData) {
```